crates/pathfinder/src/rpc.rs

```diff
@@ -220,6 +220,15 @@ Hint: If you are looking to run two instances of pathfinder, you must configure
                 .await
         },
     )?;
+    module.register_async_method("starknet_getNonce", |params, context| async move {
+        #[derive(Debug, Deserialize)]
+        pub struct NamedArgs {
+            pub contract_address: ContractAddress,
```

Comment:
I think there was discussion of pre-empting the `block_id` on this struct:

```suggestion
            #[serde(default)]
            pub block_id: Option<BlockId>,
            pub contract_address: ContractAddress,
```

This would require the query to be global state root at that block get for that address if I remember correctly.

Reply:
I opted for implementing `0.1.0` as is, instead of pre-empting the `block_id`. I think this will lead to less confusion as this is what the SDKs are expecting. It should be a simply change for `block_id` once we are there.

Probably no requirement for `pub` -- I just copied the previous registration method 🤫

Reply:
I don't think this struct can be named outside of this function, so it cannot be used as pub.